File: Clase-01/tads/lists/listADT.cpp
template <class T>
class ListADT
{
public:
  virtual T get(int pos) = 0;
  virtual void pushFront(T o) = 0;
  virtual void pushBack(T o) = 0;
  virtual void pushAt(T o, int pos) = 0;
  virtual void removeAt(int pos) = 0;
  virtual void remove(T o) = 0;
  virtual bool isEmpty() = 0;
  virtual bool contains(T o) = 0;
  virtual int size() = 0;
};

File: Clase-01/tads/lists/ListADT.java
package tads.lists;

public interface ListADT<T>{
  public T get(int pos);
  public void pushFront(T o);
  public void pushBack(T o);
  public void pushAt(T o, int pos);
  public void deleteAt(int pos);
  public void delete(T o);
  public boolean isEmpty();
  public boolean contains(T o);
  public int size();
}
